main.go

```diff
@@ -142,12 +143,26 @@ func main() { //nolint:funlen
 
 	(&webhook.OpenDataHubWebhook{}).SetupWithManager(mgr)
 
+	mgrRestConfig := mgr.GetConfig()
+	authClient := mgr.GetClient()
+	tokenReview := &authentication.TokenReview{
+		Spec: authentication.TokenReviewSpec{
+			Token: mgrRestConfig.BearerToken,
+		},
+	}
+
+	if err = authClient.Create(context.Background(), tokenReview, &client.CreateOptions{}); err != nil {
+		setupLog.Error(err, "error creating TokenReview")
+	}
+	audiences := tokenReview.Status.Audiences
```

Comment:
Two things here:

- if we don't reuse the client and config anywhere else I don't think assigning them to the variables helps
- if we log the error and don't terminate perhaps it's better to assign obtained audiences in `else` block, as it follows the intent 

```suggestion
	tokenReview := &authentication.TokenReview{
		Spec: authentication.TokenReviewSpec{
			Token: mgr.GetConfig().BearerToken,
		},
	}

	var audiences []string
	if err = mgr.GetClient().Create(context.Background(), tokenReview, &client.CreateOptions{}); err != nil {
		setupLog.Error(err, "error creating TokenReview, unable to obtain the cluster config")
	} else {
		audiences = tokenReview.Status.Audiences
	}
```

On top of that if we look at TokenReviewStatus struct it has following fields:

```
type TokenReviewStatus struct {
	// Authenticated indicates that the token was associated with a known user.
	Authenticated [bool](https://pkg.go.dev/builtin#bool)
	// User is the UserInfo associated with the provided token.
	User [UserInfo](https://pkg.go.dev/k8s.io/kubernetes/pkg/apis/authentication#UserInfo)
	// Audiences are audience identifiers chosen by the authenticator that are
	// compatible with both the TokenReview and token. An identifier is any
	// identifier in the intersection of the TokenReviewSpec audiences and the
	// token's audiences. A client of the TokenReview API that sets the
	// spec.audiences field should validate that a compatible audience identifier
	// is returned in the status.audiences field to ensure that the TokenReview
	// server is audience aware. If a TokenReview returns an empty
	// status.audience field where status.authenticated is "true", the token is
	// valid against the audience of the Kubernetes API server.
	Audiences [][string](https://pkg.go.dev/builtin#string)
	// Error indicates that the token couldn't be checked
	Error [string](https://pkg.go.dev/builtin#string)
}
```

I wonder if we also check `Authenticated` and `Error`. WDYT?

Reply:
[1374f2a](https://github.com/opendatahub-io/opendatahub-operator/pull/947/commits/1374f2a24845425f7001d66c0938b062d2c37e5b)

good idea, added a check on those fields